src/Code/serverless/Client/Select.js

```diff
@@ -6,11 +6,6 @@ const Select = class Select {
   }
 
   workflow(name) {
-    if (typeof name !== "string") {
-      throw new ExternalZenatonError(
-        `In "select.workflow()", parameter should be a string - not a "${typeof name}"`,
-      );
-    }
```

Comment:
As a side effect, this allows users to pass any type without getting a proper error message.
For example, when writing `this.select.workflow({ name: "toto" })` you won't have a useful error message. You will probably have a weird result when the object will be at some point casted to a string.

I think we can aim to keep the type checking but doing it only when the variable is provided.

Something like this maybe?

```js
    if (!!name && typeof name !== "string") {
      throw new ExternalZenatonError(
        `In "select.workflow()", parameter should be a string - not a "${typeof name}"`,
      );
    }
```

I think it will do basically what you wanted, but with type checking when name is provided.

Reply:
you are right - done